### PR TITLE
OHRM5X-2120: Add new detailed model for MyInfoAPI

### PR DESCRIPTION
**Pull request by ManulMax**

### CODE REVIEW

File: src/plugins/orangehrmPimPlugin/Api/Model/MyInfoDetailedModel.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    /**
+     * @param Employee $employee
+     */
+    public function __construct(Employee $employee)
+    {
+        $this->setEntity($employee);
+        $this->setFilters(
+            [
+                'lastName',
```
Comment (ManulMax) [inline src/plugins/orangehrmPimPlugin/Api/Model/MyInfoDetailedModel.php:62]: add empNumber

File: src/plugins/orangehrmPimPlugin/Api/MyInfoAPI.php [hunk 2]
```diff
@@ -46,7 +46,7 @@ class MyInfoAPI extends Endpoint implements ResourceEndpoint
     public const MODEL_DETAILED = 'detailed';
     public const MODEL_MAP = [
         self::MODEL_DEFAULT => EmployeeModel::class,
```
Comment (ManulMax) [inline src/plugins/orangehrmPimPlugin/Api/MyInfoAPI.php:48]: Add EmployeeDetailedModel

File: src/plugins/orangehrmPimPlugin/Api/MyInfoAPI.php [hunk 3]
```diff
@@ -44,9 +45,11 @@ class MyInfoAPI extends Endpoint implements ResourceEndpoint
     public const FILTER_MODEL = 'model';
     public const MODEL_DEFAULT = 'default';
     public const MODEL_DETAILED = 'detailed';
+    public const MODEL_SUMMERY = 'summery';
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPimPlugin/Api/MyInfoAPI.php:48]:
```suggestion
    public const MODEL_SUMMARY = 'summary';
```
typo

File: src/plugins/orangehrmPimPlugin/Api/MyInfoAPI.php [hunk 4]
```diff
@@ -59,7 +62,7 @@ class MyInfoAPI extends Endpoint implements ResourceEndpoint
      *         required=false,
      *         @OA\Schema(
      *             type="string",
-     *             enum={OrangeHRM\Pim\Api\MyInfoAPI::MODEL_DEFAULT, OrangeHRM\Pim\Api\MyInfoAPI::MODEL_DETAILED},
+     *             enum={OrangeHRM\Pim\Api\MyInfoAPI::MODEL_DEFAULT, OrangeHRM\Pim\Api\MyInfoAPI::MODEL_DETAILED, },
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPimPlugin/Api/MyInfoAPI.php:65]:
```suggestion
     *             enum={OrangeHRM\Pim\Api\MyInfoAPI::MODEL_DEFAULT, OrangeHRM\Pim\Api\MyInfoAPI::MODEL_DETAILED, OrangeHRM\Api\MyInfoAPI::MODEL_SUMMARY},
```

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**